core/src/test/resources/org/apache/calcite/test/RelOptRulesTest.xml

```diff
@@ -1028,6 +1028,39 @@ LogicalProject(MGR=[$0], SUM_SAL=[$2])
     LogicalAggregate(group=[{0, 1}], SUM_SAL=[SUM($2)])
       LogicalProject(MGR=[$3], DEPTNO=[$7], SAL=[$5])
         LogicalTableScan(table=[[CATALOG, SALES, EMP]])
+]]>
+    </Resource>
+  </TestCase>
+  <TestCase name="testAggregateUnionTransposeWithOneUniqueInput">
+    <Resource name="sql">
+      <![CDATA[select deptno, SUM(t) from (
+select deptno, 1 as t from sales.emp e1
+union all
+select distinct deptno, 2 as t from sales.emp e2)
+group by deptno]]>
+    </Resource>
+    <Resource name="planBefore">
+      <![CDATA[
+LogicalAggregate(group=[{0}], EXPR$1=[SUM($1)])
+  LogicalUnion(all=[true])
+    LogicalProject(DEPTNO=[$7], T=[1])
+      LogicalTableScan(table=[[CATALOG, SALES, EMP]])
+    LogicalAggregate(group=[{0, 1}])
+      LogicalProject(DEPTNO=[$7], T=[2])
+        LogicalTableScan(table=[[CATALOG, SALES, EMP]])
+]]>
+    </Resource>
+    <Resource name="planAfter">
+      <![CDATA[
+LogicalAggregate(group=[{0}], EXPR$1=[SUM($1)])
```

Comment:
Wait I am still confused now: if we say that all inputs have unique keys thus we push down agg, then why we still need the top agg?

Reply:
We do the opposite - return if all inputs are already unique. Otherwise - do a pushdown. Please see the previous comment for more details.

Reply:
Ok, then the pushdown make senses since it does a pre-agg before UNION and then post-agg after UNION, that gives correct result.

And if all inputs are already unique, then there is no need to push down agg. In this case, are you aware if there is a test existing to verify the return case? (I tried to search testAggregateUnionTransposeWithAllUniqueInputs or just testAggregateUnionTranspose but only found your new test case).

Reply:
I added another test:
1. `testAggregateUnionTransposeWithOneInputUnique` ensures that `Aggregate` is pushed down to all inputs because at least one input is not unique.
2. `testAggregateUnionTransposeWithAllInputsUnique` ensures that `Aggregate` is not pushed because all inputs are already unique (the "return" branch).